src/lib/assetBridger/l1l3Bridger.ts

```diff
@@ -0,0 +1,1166 @@
+  }
+
+  /**
+   * Get the L2 token contract at the provided address
+   * Note: This function just returns a typed ethers object for the provided address, it doesnt
+   * check the underlying form of the contract bytecode to see if it's an erc20, and doesn't ensure the validity
+   * of any of the underlying functions on that contract.
+   */
+  public getL2TokenContract(
+    l2TokenAddr: string,
+    l2Provider: Provider
+  ): L2GatewayToken {
+    return L2GatewayToken__factory.connect(l2TokenAddr, l2Provider)
+  }
+
+  /**
+   * Get the L3 token contract at the provided address
+   * Note: This function just returns a typed ethers object for the provided address, it doesnt
+   * check the underlying form of the contract bytecode to see if it's an erc20, and doesn't ensure the validity
+   * of any of the underlying functions on that contract.
+   */
+  public getL3TokenContract(
+    l3TokenAddr: string,
+    l3Provider: Provider
+  ): L2GatewayToken {
+    return L2GatewayToken__factory.connect(l3TokenAddr, l3Provider)
+  }
+
+  /**
+   * Whether the L1 token has been disabled on the router given an L1 token address
+   */
+  public async l1TokenIsDisabled(
+    l1TokenAddress: string,
+    l1Provider: Provider
+  ): Promise<boolean> {
+    await this._checkL1Network(l1Provider)
+
+    return this._tokenIsDisabled(
+      l1TokenAddress,
+      this.l2Network.tokenBridge.l1GatewayRouter,
+      l1Provider
+    )
+  }
+
+  /**
+   * Whether the L2 token has been disabled on the router given an L2 token address
+   */
+  public async l2TokenIsDisabled(
+    l2TokenAddress: string,
+    l2Provider: Provider
+  ): Promise<boolean> {
+    await this._checkL2Network(l2Provider)
+
+    return this._tokenIsDisabled(
+      l2TokenAddress,
+      this.l3Network.tokenBridge.l1GatewayRouter,
+      l2Provider
+    )
+  }
+
+  /**
+   * Given L2Forwarder parameters, get the address of the L2Forwarder contract
+   */
+  public async l2ForwarderAddress(
+    params: L2ForwarderPredictor.L2ForwarderParamsStruct,
+    l2Provider: Provider
+  ): Promise<string> {
+    await this._checkL2Network(l2Provider)
+
+    return L2ForwarderFactory__factory.connect(
+      this.teleporterAddresses.l2ForwarderFactory,
+      l2Provider
+    ).l2ForwarderAddress(params)
+  }
+
+  /**
+   * Given a parent network token bridge deposit transaction receipt, get the recipient address on the child network
+   */
+  public getRecipientFromParentBridgeTx(
+    depositTxReceipt: L1ContractCallTransactionReceipt
+  ) {
+    const topic0 =
+      L1GatewayRouter__factory.createInterface().getEventTopic('TransferRouted')
+    const log = depositTxReceipt.logs.find(x => x.topics[0] === topic0)
+
+    if (!log) {
+      throw new ArbSdkError(`Could not find TransferRouted event in tx receipt`)
+    }
+
+    // topic 3 is "_userTo"
+    const bytes32UserTo = log.topics[3]
+
+    // parse address
+    return ethers.utils.getAddress(bytes32UserTo.slice(26))
```

Comment:
I think you can alternatively use 

getAddress(hexDataSlice(bytes32UserTo, 12))